space_packet_parser/xtcedef.py

```diff
@@ -969,7 +969,27 @@ def read(self, format_string: str, update_position: bool = True) -> Union[int, f
         : Union[int, float, str, bytes]
             Value read from the packet data according to the format specifier.
         """
-        # pylint: disable=too-many-branches
+        # pylint: disable=too-many-branches, too-many-return-statements
+
+        def _twos_complement(val: int, bit_width: int):
```

Comment:
Were you able to profile this at all? I'm slightly concerned about defining the function repeatedly in here each time through the loop, so it might be beneficial to put it in the outer module scope.

Reply:
This doesn't seem to make a ton of difference, so no need to fix this up now.